# Printing Hello World Program

In [3]:
print("hello world,1,1.5,6")

hello world,1,1.5,6


In [6]:
print("hello,hi",1,2,3,4,sep = "/")

hello,hi/1/2/3/4


In [9]:
print("hello world")
print("my world")

hello world
my world


In [17]:
print("hello world",end = " ")
print("hi maria")
print("hi nancy")

hello world hi maria
hi nancy


In [18]:
print("hello world",end = " ")
print("hi maria", end = " ")
print("hi nancy")

hello world hi maria hi nancy


# 2)Data types

In [20]:
# integer values
print(8)
# max number that can alllowee in python
print(1e309)

8
inf


In [23]:
# float values
print(4.6)
print(1e308)

4.6
1e+308


In [25]:
# boolean
print(True)
print(False)

True
False


In [26]:
# String
print('hello')

hello


In [27]:
# complex number
print('5+6j')

5+6j


In [28]:
# List array
# and in the c kanguage it will be like
print([1,3,4,5,6])

[1, 3, 4, 5, 6]


In [29]:
# Tuple
print("hi sana")

hi sana


In [31]:
# Sets
print({1})

{1}


In [33]:
# Dictionary
print({"name":"maria", "surname":"bravo"})

{'name': 'maria', 'surname': 'bravo'}


# 3 .Variable

In [34]:
a =6 
b = 7
print(a+b)

13


## interview important

### Dynamic typing

in python, if more your python interpreter is intelligent that he detect the what kind of variable your declaring there is no need as like as c and c++ you declare the data type frist

In [37]:
# e.g
name = "maria sharapoa"
print(name)

maria sharapoa


## Dynamic binding

in python, there is no fix variable datatypeit hold any kind of data type thorugh out code. value you can modified as per you want but in C or C++ you follow the static binding where as onc you delcaare the data types then  you can change in the whole code. yes you can modified the variable but yyou cant the data type that alocate you taht variable

In [36]:
#e.g in python
a=5
print(a)
a="heena"
print(a)

#e.g in c/c++

int a = 5

5
heena


### Fun part of python programming the variabe data type allocation

In [38]:
a,b,c = 1,2,3
print(a,b,c)

1 2 3


In [39]:
# you can give same value to different different variable 
a=b=c = 5
print(a,b,c)

5 5 5


# 4)comment

In [40]:
# this is the comment single line comment
'''this is multiline comment
thhis second line'''

'this is multiline comment\nthhis second line'

# 5)Keywords and Identifiers


In [46]:
#Keywords
# ther are 35 keywords
# this is an reserve keywords youc cannot use this reserve keywords to delcare the variables
from keyword import kwlist

In [48]:
kwlist

['False',
 'None',
 'True',
 'and',
 'as',
 'assert',
 'async',
 'await',
 'break',
 'class',
 'continue',
 'def',
 'del',
 'elif',
 'else',
 'except',
 'finally',
 'for',
 'from',
 'global',
 'if',
 'import',
 'in',
 'is',
 'lambda',
 'nonlocal',
 'not',
 'or',
 'pass',
 'raise',
 'return',
 'try',
 'while',
 'with',
 'yield']

In [49]:
# cehck the length of keywords
len(kwlist)

35

# 6)Identifiers

### Identifiers is like whatever you name provde to any variable,class or object the identity of that perticular object that can say the Identifiers

#### Rules of Identifeirs

* you can not start with digit
* in special character youc an only use the _


# 7)User Input

##### thier sre 2 types of software
* Stactic
* Dynamic

In [51]:
input("Enter email")

Enter email bro@gmail.com


'bro@gmail.com'

In [52]:
# take input from users and store them in a variable
fnum = input("enter your first num")
snum = input("enter second num")

# add the 2 variables
print(fnum+snum)



enter your first num 56
enter second num 67


5667


#### ther re one question for you tell me the reason behind the why hasbeen print(5667) instead of 56+67 = 123
ans :- "when input taken that time you have to declare the what kind of data type input you want if you not declare then that time is will be taken unviersal data type String that why it added the 5667"

In [53]:
# Correct way
fnum = int(input("enter your first num"))
snum = int(input("enter second num"))

# add the 2 variables
print(fnum+snum)


enter your first num 56
enter second num 67


123


# 8)Type Convertion

##### there are 2 types 
* Implicit 
* Explicit

In [57]:
# Implicit e.g
print(5+5.6)
print(type(5),type(5.6))
# see the type there has been 5 that has been int type and 5.6 is an float data type this has an diferent data types but they still add 
# by another programming law c & c++ is has been ocurred error but in python it ha san intelligance that enough to understand 5+5.6 will be 10.6
# that implicit examples

10.6
<class 'int'> <class 'float'>


In [58]:
# Explicit examples
print(4+"4")
# why error has been ocurred beacuse of python dont know that in real life this is possible that you add int +string value. so when you request personally 
# cahnge the data type of the of given value that is an exlpicit example

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [60]:
#Explicit examples
print(int("4"))
print(type(int(4)))

4
<class 'int'>


# NOTES:- Python types conversion does not change original data it create new value of the object and work on that

In [65]:
# E.g:

fnum = input("enter your first num")
snum = input("enter second num")

# add the 2 variables
result = int(fnum)+int(snum)
print(result)

# lets check the data type
print(type(fnum))
print("---------------------------")
# Correct way
fnum = int(input("enter your first num"))
snum = int(input("enter second num"))

# add the 2 variables
print(fnum+snum)
type(fnum)


enter your first num 56
enter second num 67


123
<class 'str'>
---------------------------


enter your first num 56
enter second num 67


123


int

# 9) Literals

##### Variables is an cointainer where you stored the value in that 
##### Literals means the values stored in the variable taht is called as literals

In [69]:
a = 0b1010 #Binary Literals
b = 100 #Decimal Literal 
c = 0o310 #Octal Literal
d = 0x12c #Hexadecimal Literal

#Float Literal
float_1 = 10.5 
float_2 = 1.5e2 # 1.5 * 10^2
float_3 = 1.5e-3 # 1.5 * 10^-3

#Complex Literal 
x = 3.14j

print(a, b, c, d)
print(float_1, float_2,float_3)
print(x, x.imag, x.real)

10 100 200 300
10.5 150.0 0.0015
3.14j 3.14 0.0


In [70]:
string = 'This is Python'
strings = "This is Python"
char = "C"
multiline_str = """This is a multiline string with more than one line code."""
unicode = u"\U0001f600\U0001F606\U0001F923"
raw_str = r"raw \n string"

print(string)
print(strings)
print(char)
print(multiline_str)
print(unicode)
print(raw_str)

This is Python
This is Python
C
This is a multiline string with more than one line code.
😀😆🤣
raw \n string


In [71]:
a = True + 4
b = False + 10

print("a:", a)
print("b:", b)

a: 5
b: 10


In [72]:
k = None
a = 5
b = 6
print('Program exe')

Program exe
